dataframe-excel/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/xlsx.kt

```diff
@@ -246,9 +246,9 @@ public fun <T> DataFrame<T>.writeExcel(
     columnsSelector: ColumnsSelector<T, *> = { all() },
     sheetName: String? = null,
     writeHeader: Boolean = true,
-    factory: () -> Workbook
+    factory: Workbook
```

Comment:
Let's remove this overload. I think i wanted to create an alias for `() -> Workbook` so that people could call this function like `df.writeExcel(os, ..., WorkbookFactory.XLSX)`, and `WorkbookFactory.XLSX` here is a class owned by the dataframe. But i shelved the idea, so this overload adds no value compared to the one below

Reply:
Looking back at it, should it be renamed to workBook now? Since it is what is really getting passed after this change. I can change it if so.